### PR TITLE
options in query

### PR DESCRIPTION
**Pull request by beowulfenator**
This is regarding issue #70.

### CODE REVIEW

File: Query.php [hunk 1]
```diff
@@ -157,6 +157,10 @@ class Query extends Component implements QueryInterface
      * @see http://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-min-score.html
      */
     public $minScore;
+    /**
+     * @var array list of options that will passed to commands created by this query.
+     */
```
Comment (cebe) [inline Query.php]: need `@since` annotation.

File: Query.php [hunk 2]
```diff
@@ -607,4 +611,20 @@ public function minScore($minScore)
         $this->minScore = $minScore;
         return $this;
     }
+
+    /**
+     * Sets the options to be passed to the command created by this query.
+     * @param array $options the options to be set.
+     * @return $this the query object itself
+     */
```
Comment (cebe) [inline Query.php]: same here.

File: Query.php [hunk 3]
```diff
@@ -607,4 +611,20 @@ public function minScore($minScore)
         $this->minScore = $minScore;
         return $this;
     }
+
+    /**
+     * Sets the options to be passed to the command created by this query.
```
Comment (cebe) [inline Query.php]: should be more clear about what kind of options these are, e.g. `options that are appended to the query URL such as preference, ... etc...`

**Comment by cebe** [comment]
Travis CI fails, maybe the version does not support the options? https://travis-ci.org/yiisoft/yii2-elasticsearch/jobs/112157332#L47

**Comment by beowulfenator** [comment]
I'm glad you agree with this in general. I'll add more docs, tests, and probably the `addOptions()` method. I wonder why it's failing the test, though. Will investigate.

**Comment by beowulfenator** [comment]
The tests were failing, because the option that I added as a test was not supported in older versions of ES. I've changed the test to use another option, and all is well now.

@SilverFire, I've fixed tests for minScore. Minimum score is supported in all versions of ES, but to test it we need to explicitly assign scores to test results, which I do using `script_score`.

@cebe, any other suggestions?

File: Query.php [hunk 4]
```diff
@@ -607,4 +613,41 @@ public function minScore($minScore)
...
+     * @param array $options the options to be set.
+     * @return $this the query object itself
+     * @throws \yii\base\InvalidParamException if $options is not an array
+     * @see Command::$options
+     * @since  2.0.4
+     */
+    public function options($options)
+    {
+        if (!is_array($options)) {
+            throw new \yii\base\InvalidParamException('Array parameter expected, ' . gettype($options) . ' received.');
```
Comment (cebe) [inline Query.php]: `\yii\base\InvalidParamException` should be included with `use`.

**Comment by cebe** [comment]
If you fix the exception class `use` and add a CHANGELOG line it is good to merge imo

**Comment by cebe** [comment]
:+1:

File: Query.php [hunk 5]
```diff
@@ -157,6 +158,12 @@ class Query extends Component implements QueryInterface
      * @see http://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-min-score.html
      */
     public $minScore;
+    /**
+     * @var array list of options that will passed to commands created by this query.
```
Comment (SilverFire) [inline Query.php]: `will passed` -> `will be passed`

**Pull request merged**